### ISSUE TITLE
The model.fit module reported an error

### ISSUE THREAD

**Issue by 17860612038** [keras-team/keras#18173]
1.Code presentation
#one
import tensorflow as tf
import numpy as np
import matplotlib.pyplot as plt
import h5py
import tensorflow.keras
import os, random, shutil
 
 
#two
 
base_dir = 'D:\CNN\chuli'
 
train_dir = os.path.join(base_dir, 'train/')  
validation_dir = os.path.join(base_dir, 'validation/')  
 
train_BT_dir = os.path.join(train_dir, 'BT')  # directory with our training BT pictures
train_HP_dir = os.path.join(train_dir, 'HP')  # directory with our training HP pictures
 
validation_BT_dir = os.path.join(validation_dir, 'BT')  # directory with our validation BT pictures
validation_HP_dir = os.path.join(validation_dir, 'HP')  # directory with our validation HP pictures
 
PATH1='D:\\CNN\\chuli\\train//BT'
PATH2='D:\\CNN\\chuli\\train//HP'
 
PATH7='D:\\CNN\\chuli\\validation//BT'
PATH8='D:\\CNN\\chuli\\validation//HP'
 
num_BT_train = len(os.listdir(PATH1))
num_HP_train = len(os.listdir(PATH2))
 
num_BT_validation = len(os.listdir(PATH7))
num_HP_validation = len(os.listdir(PATH8))
 
total_train = num_BT_train + num_HP_train
total_validation=num_BT_validation + num_HP_validation 
 
print(total_train)
print(total_validation) 
 
#three
batch_size =10,
epochs =10,
IMG_HEIGHT =2048,
IMG_WIDTH = 2048,
 
 
#four
train_image_generator = tf.keras.preprocessing.image.ImageDataGenerator(rescale=1. / 255)
test_image_generator = tf.keras.preprocessing.image.ImageDataGenerator(rescale=1. / 255)
validation_image_generator = tf.keras.preprocessing.image.ImageDataGenerator(rescale=1. / 255)
 
#five
 
train_data_gen = train_image_generator.flow_from_directory(batch_size=batch_size,
                                                           directory=train_dir,
                                                           shuffle=True,
                                                           target_size=(IMG_HEIGHT, IMG_WIDTH),
                                                           class_mode='binary')
 
val_data_gen = validation_image_generator.flow_from_directory(batch_size=batch_size,
                                                              directory=validation_dir,
                                                              target_size=(IMG_HEIGHT, IMG_WIDTH),
                                                              class_mode='binary')
 
#six
import tensorflow as tf
IMG_HEIGHT =2048
IMG_WIDTH = 2048
model=tf.keras.Sequential([
    tf.keras.layers.Conv2D(512,3, padding='same', activation='relu', input_shape=(IMG_HEIGHT, IMG_WIDTH, 3)),
    tf.keras.layers.MaxPooling2D(),
    tf.keras.layers.Conv2D(256,3, padding='same', activation='relu'),
    tf.keras.layers.MaxPooling2D(),
    tf.keras.layers.Conv2D(128,3, padding='same', activation='relu'),
    tf.keras.layers.MaxPooling2D(),
    tf.keras.layers.Dense(512, activation='relu'),
    tf.keras.layers.Dense(1,activation='sigmoid')
    ])
 
#seven
model.compile(optimizer= 'Adam',
              loss='CategoricalCrossentropy',
              metrics= 'accuracy')
model.summary()
 
 
#eight
history = model.fit_generator(
    train_data_gen,
    steps_per_epoch=total_train//(batch_size,
    epochs==3，
    validation_data=val_data_gen,
    validations_steps=total_val//batch_size
)
2.Error presentation
![image](https://github.com/keras-team/keras/assets/97085347/6a4bb5cd-2668-4b4a-a9b6-d98153ebb4c1)
3.Environmental display
tensorflow GPU=2.6.0
CUDA11.2,
CUDNN8.1
GTX970,4G

**Comment by tilakrayal** [comment]
@17860612038,
I was facing a different issue while executing the mentioned code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/3562b88636de8086763dd675615d539f/untitled1184.ipynb) and provide the complete dependencies to reproduce the issue. 

Also **Model.fit_generator** is deprecated starting from tensorflow 2.1.0. Please find the documentation here: https://www.tensorflow.org/versions/r2.1/api_docs/python/tf/keras/Model#fit

As you can see the first argument of the Model.fit can take a generator so just pass it to your generator.
https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit. Thank you!

**Comment by 17860612038** [comment]
> @17860612038, I was facing a different issue while executing the mentioned code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/3562b88636de8086763dd675615d539f/untitled1184.ipynb) and provide the complete dependencies to reproduce the issue.
> 
> Also **Model.fit_generator** is deprecated starting from tensorflow 2.1.0. Please find the documentation here: https://www.tensorflow.org/versions/r2.1/api_docs/python/tf/keras/Model#fit
> 
> As you can see the first argument of the Model.fit can take a generator so just pass it to your generator. https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit. Thank you!

Thank you for your reply. The information you provided is of great help to me

**Comment by tilakrayal** [comment]
@17860612038,
Glad the suggestion helped you. Could you please feel free to move this issue to closed status. Thank you!
